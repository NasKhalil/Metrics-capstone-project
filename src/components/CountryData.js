import React from 'react';
import PropTypes from 'prop-types';
import { BsArrowRightCircleFill } from 'react-icons/bs';

import virus from '../images/virus.png';
import '../styling/CountryData.css';

const CountryData = ({ name, confirmedCases }) => (
  <li className="country-data">
    <div className="right">
      <BsArrowRightCircleFill className="arrow" />
    </div>
    <div className="country">
      <img src={virus} alt="Covid logo" className="covid-img" />
      <aside>
        <h2>{name}</h2>
        <span>Total Confirmed</span>
        <h3>{`${confirmedCases} cases`}</h3>
      </aside>
    </div>
  </li>
);

CountryData.propTypes = {
  name: PropTypes.string.isRequired,
  confirmedCases: PropTypes.number.isRequired,
};

export default CountryData;
