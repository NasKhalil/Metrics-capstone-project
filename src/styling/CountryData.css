.country-data {
  list-style: none;
  width: 80%;
  padding: 30px 10px;
  margin: 0 auto;
  line-height: 1.7;
}

.right {
  display: flex;
  justify-content: flex-end;
  align-items: center;
}

.arrow {
  font-size: 18px;
}

.country {
  display: flex;
  justify-content: space-around;
  align-items: center;
}

.covid-img {
  width: 30px;
  opacity: 0.8;
}

.country-data aside {
  letter-spacing: 4px;
}

.country-data aside h2 {
  font-size: 16px;
}

.country-data aside h3 {
  font-size: 12px;
}

@media all and (min-width: 1024px) {
  .country-data {
    padding: 30px 0;
  }

  .country {
    text-align: start;
    display: flex;
    justify-content: space-between;
    align-items: baseline;
    margin-top: 15px;
  }

  .covid-img {
    width: 20%;
  }

  .arrow {
    font-size: 21px;
  }

  .country-data aside h2 {
    font-size: 27px;
  }

  .country-data aside h3 {
    font-size: 21px;
  }
}
