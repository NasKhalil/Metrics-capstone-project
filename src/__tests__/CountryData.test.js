import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import CountryData from '../components/CountryData';

describe('CountryData test', () => {
  test('should render', () => {
    const footer = render(
      <Router>
        <Provider store={store}>
          <CountryData name="France" confirmedCases={50} />
        </Provider>
      </Router>,
    );

    expect(footer).toMatchSnapshot();
  });
});
