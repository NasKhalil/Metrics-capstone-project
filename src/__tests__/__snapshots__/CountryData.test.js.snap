// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryData test should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="country-data"
      >
        <div
          class="right"
        >
          <svg
            class="arrow"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
            />
          </svg>
        </div>
        <div
          class="country"
        >
          <img
            alt="Covid logo"
            class="covid-img"
            src="virus.png"
          />
          <aside>
            <h2>
              France
            </h2>
            <span>
              Total Confirmed
            </span>
            <h3>
              50 cases
            </h3>
          </aside>
        </div>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="country-data"
    >
      <div
        class="right"
      >
        <svg
          class="arrow"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
          />
        </svg>
      </div>
      <div
        class="country"
      >
        <img
          alt="Covid logo"
          class="covid-img"
          src="virus.png"
        />
        <aside>
          <h2>
            France
          </h2>
          <span>
            Total Confirmed
          </span>
          <h3>
            50 cases
          </h3>
        </aside>
      </div>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
